File: aoj/volume0/n0009/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        boolean isPrime[] = new boolean[1000000];
        int numberOfPrime[] = new int[1000000];

        isPrime[0] = false;
        isPrime[1] = false;

        for (int i = 2; i < 1000000; i++) {
            isPrime[i] = true;
        }

        Arrays.fill(numberOfPrime, 0);

        for (int i = 2; i < 1000000; i++) {

            if (isPrime[i] == true) {

                numberOfPrime[i] = numberOfPrime[i - 1] + 1;

                for (int j = i + i; j < 1000000; j += i) {
                    isPrime[j] = false;
                }

            } else {

                numberOfPrime[i] = numberOfPrime[i - 1];

            }

        }

        Scanner sc = new Scanner(System.in);

        while (sc.hasNextInt()) {
            System.out.println(numberOfPrime[sc.nextInt()]);
        }

    }

}

File: aoj/volume0/n0009/Main.cpp
#include <iostream>

using namespace std;

int main() {

    bool isPrime[1000000];

    for (int i = 2; i < 1000000; i++) {
        isPrime[i] = true;
    }

    for (int i = 2; i * i < 1000000; i++) {

        if (isPrime[i]) {

            for (int j = i * i; j < 1000000; j += i) {
                isPrime[j] = false;
            }

        }

    }

    int numberOfPrime[1000000] = {0};

    for (int i = 1; i < 1000000; i++) {

        if (isPrime[i]) {
            numberOfPrime[i] = numberOfPrime[i - 1] + 1;
        } else {
            numberOfPrime[i] = numberOfPrime[i - 1];
        }

    }

    int n;

    while (cin >> n) {
        cout << numberOfPrime[n] << endl;
    }

    return 0;

}
